src/pluto/pluto.py

```diff
@@ -521,45 +521,39 @@ def create_outlier(df, v1, v2, condo, mapped):
         outlier_records = outlier.to_dict("records")
         v1_outlier = [i["outlier"] for i in outlier_records if i["v"] == v1][0]
 
-        def display_dataframe(v1_outlier, field):
+        def fetch_dataframe(v1_outlier, field):
             records = [i["values"] for i in v1_outlier if i["field"] == field][0]
             if records:
                 df = pd.DataFrame(records)
                 if field == 'building_area_increase':
                     df=df.drop(columns=['pair'])
                 df['bbl']=pd.to_numeric(df['bbl'], downcast='integer')
-                return df,df.shape[0]
+                return df
             else:
-                return 'This is no outlier.'
-        
-        version_pair = v1 + '-' + v2
+                return pd.DataFrame()
+
+        def display_dataframe(v1_outlier, field):
```

Comment:
This is a great start to the refactor. I was thinking we could go further though. The markdown header, the dataframe, and info should all come from a single function call. This function should take just the name of the report and get the text for the markdown and the info from a dictionary or function.

I think this would make the code harder to read but more maintainable going forward. I think we want each section of the outlier section to look similar, so they all change the same way going forward.

Reply:
okay, that makes sense. will make the change.

Reply:
make this change by adding two dictionaries for markdown and description. If we are going to add new sections for outlier report in the future, that would be easier to modify the code.

Reply:
🔥 this is exactly what I had in mind